tests/components/blebox/test_switch.py

```diff
@@ -0,0 +1,386 @@
+
+    def turn_on0():
+        feature_mocks[0].is_on = True
+
+    feature_mocks[0].async_turn_on = AsyncMock(side_effect=turn_on0)
+    await hass.services.async_call(
+        "switch", SERVICE_TURN_ON, {"entity_id": entity_ids[0]}, blocking=True,
+    )
+
+    assert hass.states.get(entity_ids[0]).state == STATE_ON
+    assert hass.states.get(entity_ids[1]).state == STATE_OFF
+
+
+async def test_switchbox_d_second_on(switchbox_d, hass, config):
+    """Test turning switch on."""
+
+    feature_mocks, entity_ids = switchbox_d
+
+    def initial_update0():
+        feature_mocks[0].is_on = False
+        feature_mocks[1].is_on = False
+
+    feature_mocks[0].async_update = AsyncMock(side_effect=initial_update0)
+    feature_mocks[1].async_update = AsyncMock()
+    await async_setup_entities(hass, config, entity_ids)
+    feature_mocks[0].async_update = AsyncMock()
+
+    def turn_on1():
+        feature_mocks[1].is_on = True
+
+    feature_mocks[1].async_turn_on = AsyncMock(side_effect=turn_on1)
+    await hass.services.async_call(
+        "switch", SERVICE_TURN_ON, {"entity_id": entity_ids[1]}, blocking=True,
+    )
+
+    assert hass.states.get(entity_ids[0]).state == STATE_OFF
+    assert hass.states.get(entity_ids[1]).state == STATE_ON
+
+
+async def test_switchbox_d_first_off(switchbox_d, hass, config):
+    """Test turning switch on."""
+
+    feature_mocks, entity_ids = switchbox_d
+
+    def initial_update_any():
+        feature_mocks[0].is_on = True
+        feature_mocks[1].is_on = True
+
+    feature_mocks[0].async_update = AsyncMock(side_effect=initial_update_any)
+    feature_mocks[1].async_update = AsyncMock()
+    await async_setup_entities(hass, config, entity_ids)
+    feature_mocks[0].async_update = AsyncMock()
+
+    def turn_off0():
+        feature_mocks[0].is_on = False
+
+    feature_mocks[0].async_turn_off = AsyncMock(side_effect=turn_off0)
+    await hass.services.async_call(
+        "switch", SERVICE_TURN_OFF, {"entity_id": entity_ids[0]}, blocking=True,
+    )
+
+    assert hass.states.get(entity_ids[0]).state == STATE_OFF
+    assert hass.states.get(entity_ids[1]).state == STATE_ON
+
+
+async def test_switchbox_d_second_off(switchbox_d, hass, config):
+    """Test turning switch on."""
+
+    feature_mocks, entity_ids = switchbox_d
+
+    def initial_update_any():
+        feature_mocks[0].is_on = True
+        feature_mocks[1].is_on = True
+
+    feature_mocks[0].async_update = AsyncMock(side_effect=initial_update_any)
+    feature_mocks[1].async_update = AsyncMock()
+    await async_setup_entities(hass, config, entity_ids)
+    feature_mocks[0].async_update = AsyncMock()
+
+    def turn_off1():
+        feature_mocks[1].is_on = False
+
+    feature_mocks[1].async_turn_off = AsyncMock(side_effect=turn_off1)
+    await hass.services.async_call(
+        "switch", SERVICE_TURN_OFF, {"entity_id": entity_ids[1]}, blocking=True,
+    )
+    assert hass.states.get(entity_ids[0]).state == STATE_ON
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use of assert detected. The enclosed code will be removed when compiling to optimised byte code.](https://app.codacy.com/manual/foreign-sub/home-assistant/pullRequest?prid=5535891)